File: 344.reverse-string/reverse-string.py
class Solution(object):
    def reverseString(self, s):
        """
        :type s: str
        :rtype: str
        """
        lists = list(s)
        for i in range(len(s)/2):
            temp = lists[i]
            lists[i] = lists[len(s) -1 - i]
            lists[len(s) -1 -i] = temp
        return ''.join(lists)


1. list(str): convert string to a list of char
2. ''.join(lists): join the char together with '', will actually convert it back to string

File: 344.reverse-string/reverse-string.java
1. attach char from the end of the string to create reverse string --- time limit exceeded 
var reverseString = function(s) {
    array = s.split();
    start = 0;
    end = array.length - 1;
    while(start < end){
        temp = array[start];
        array[start] = array[end];
        array[end] = temp;
        start++;
        end--;
    }
    return array.join();
};

2. convert string to a char array, create two points to start and end, exchange the char, convert it back to string as result
public class Solution {
    public String reverseString(String s) {
        char[] array = s.toCharArray();
        int start = 0;
        int end = s.length() - 1;
        while (start<end){
            char temp = array[start];
            array[start] = array[end];
            array[end] = temp;
            start++;
            end--;
        }
        return new String(array);
    }
}
